src/azure-cli/azure/cli/command_modules/acs/_validators.py

```diff
@@ -151,6 +151,15 @@ def validate_linux_host_name(namespace):
                        'letters, numbers, or dashes (-).')
 
 
+def validate_snapshot_name(namespace):
+    """Validates a nodepool snapshot name to be alphanumeric and dashes."""
+    rfc1123_regex = re.compile(r'^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$')  # pylint:disable=line-too-long
+    found = rfc1123_regex.findall(namespace.snapshot_name)
+    if not found:
+        raise CLIError('--name cannot exceed 63 characters and can only contain '
+                       'letters, numbers, or dashes (-).')
```

Comment:
Please use the specific error type (such as the `InvalidArgumentValueError`) instead of `CLIError`

Reply:
changed to InvalidArgumentValueError